lib/parser.php

```diff
@@ -0,0 +1,161 @@
+<?php
+/**
+ * Includes revisions to the block parser not yet proposed for core adoption.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Filters to override default block parser with Gutenberg's custom
+ * implementation.
+ *
+ * @since 6.9.0
+ */
+function gutenberg_replace_block_parser_class() {
+	return 'WP_Sourced_Attributes_Block_Parser';
+}
+add_filter( 'block_parser_class', 'gutenberg_replace_block_parser_class' );
+
+/**
+ * Given a registered block type settings array, assigns default attributes.
+ * This must be called manually, as there is currently no way to hook to block
```

Comment:
Now, that `register_block_type_args` filter was introduced with https://core.trac.wordpress.org/ticket/49615, we can add this functionality in WordPress core. I will extract this function and propose it as an enhancement to the registration process on the server.

Reply:
The proposal is ready at https://github.com/WordPress/wordpress-develop/pull/383.